addons/mrp/models/mrp_bom.py

```diff
@@ -157,7 +190,13 @@ def check_cycle(v, visited, recStack, graph):
                 continue
 
             line_quantity = current_qty * current_line.product_qty
-            bom = self._bom_find(product=current_line.product_id, picking_type=picking_type or self.picking_type_id, company_id=self.company_id.id)
```

Comment:
AFAICS this patch would render `_bom_find` useless, so I'm not sure it would fit fine in a stable branch. #33967 is less aggressive for a stable branch IMHO.

If we target this one to master, then possibly it's not aggressive enough, and the `_bom_find` method should be removed, as it is mostly useless now.